### ISSUE TITLE
Re-Training OpenPifPaf with Custom Datasets and Get Negative Losses

### ISSUE THREAD

**Issue by yyccli** [openpifpaf/openpifpaf#493]
Hi @svenkreiss 
I'm new to openpifpaf and i want to re-train openpifpaf with my own dataset. My dataset is quite similar to CrowdPose so i followed the code in plugins -> crowdpose -> module.py to build my dataset module.
I use the  hyper-parameters that you trained crowdpose before as my hyper-parameters. The problem is that my head losses become negative and thus leading my total losses become negative.
I know this problem is similar to #327 , but my negative losses become very large.

Here is my cli when training:

```
CUDA_VISIBLE_DEVICES=0,1,2 python3 -m openpifpaf.train \
--lr=0.00001 --momentum=0.98 --b-scale=10.0 \
--epochs=300 --val-interval=10 --lr-decay 250 280 --lr-decay-epochs=10 \
--batch-size=16 \
--weight-decay=1e-5 \
--dataset=GTAHPE_CrowdPose_v1 --gtahpe-cp-v1-upsample=2 --gtahpe-cp-v1-orientation-invariant=0.1 \
--basenet=resnet50
```

where `GTAHPE_CrowdPose_v1` is my own dataset module

The logs might be:

```
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:54:08,097", "type": "train", "epoch": 138, "batch": 0, "n_batches": 1080, "time": 0.797, "data_time": 9.736, "lr": 1e-05, "loss": -1296.194, "head_losses": [225.899, -512.943, 8.379, 212.05, -701.879, -538.634, 5.47, 5.463]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:54:15,787", "type": "train", "epoch": 138, "batch": 11, "n_batches": 1080, "time": 0.727, "data_time": 0.0, "lr": 1e-05, "loss": -2209.859, "head_losses": [331.23, -956.454, 13.434, 296.603, -1071.098, -841.242, 8.842, 8.826]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:54:23,558", "type": "train", "epoch": 138, "batch": 22, "n_batches": 1080, "time": 0.623, "data_time": 0.0, "lr": 1e-05, "loss": -639.061, "head_losses": [256.395, -282.576, 9.381, 233.921, -514.675, -353.724, 6.113, 6.104]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:54:31,039", "type": "train", "epoch": 138, "batch": 33, "n_batches": 1080, "time": 0.686, "data_time": 0.0, "lr": 1e-05, "loss": -929.859, "head_losses": [265.765, -371.642, 8.5, 216.449, -656.908, -404.752, 6.369, 6.36]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:54:38,295", "type": "train", "epoch": 138, "batch": 44, "n_batches": 1080, "time": 0.674, "data_time": 0.0, "lr": 1e-05, "loss": -1491.17, "head_losses": [439.163, -655.09, 14.622, 411.986, -999.84, -722.109, 10.06, 10.039]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:54:46,142", "type": "train", "epoch": 138, "batch": 55, "n_batches": 1080, "time": 0.686, "data_time": 0.0, "lr": 1e-05, "loss": -300.228, "head_losses": [371.934, -262.61, 11.384, 332.636, -488.329, -281.838, 8.295, 8.301]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:54:53,010", "type": "train", "epoch": 138, "batch": 66, "n_batches": 1080, "time": 0.392, "data_time": 0.0, "lr": 1e-05, "loss": -675.62, "head_losses": [143.893, -236.188, 6.266, 138.831, -444.047, -292.95, 4.294, 4.282]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:55:00,431", "type": "train", "epoch": 138, "batch": 77, "n_batches": 1080, "time": 0.668, "data_time": 0.0, "lr": 1e-05, "loss": -541.973, "head_losses": [277.865, -292.349, 8.111, 243.225, -487.13, -303.674, 5.989, 5.99]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:55:05,517", "type": "train", "epoch": 138, "batch": 88, "n_batches": 1080, "time": 0.39, "data_time": 0.0, "lr": 1e-05, "loss": -645.924, "head_losses": [178.856, -333.206, 4.471, 173.769, -399.931, -277.066, 3.597, 3.585]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:55:09,818", "type": "train", "epoch": 138, "batch": 99, "n_batches": 1080, "time": 0.39, "data_time": 0.0, "lr": 1e-05, "loss": -811.067, "head_losses": [319.767, -392.054, 8.991, 269.024, -595.256, -433.782, 6.122, 6.122]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:55:14,118", "type": "train", "epoch": 138, "batch": 110, "n_batches": 1080, "time": 0.389, "data_time": 0.0, "lr": 1e-05, "loss": -1371.847, "head_losses": [232.665, -590.196, 7.957, 233.592, -734.011, -533.664, 5.91, 5.9]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:55:19,589", "type": "train", "epoch": 138, "batch": 121, "n_batches": 1080, "time": 0.726, "data_time": 0.0, "lr": 1e-05, "loss": -1104.89, "head_losses": [548.463, -708.017, 9.709, 459.955, -831.94, -597.062, 7.007, 6.995]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:55:26,850", "type": "train", "epoch": 138, "batch": 132, "n_batches": 1080, "time": 0.665, "data_time": 0.0, "lr": 1e-05, "loss": -825.392, "head_losses": [321.409, -421.86, 7.887, 255.939, -575.858, -424.249, 5.676, 5.663]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:55:34,667", "type": "train", "epoch": 138, "batch": 143, "n_batches": 1080, "time": 0.679, "data_time": 0.0, "lr": 1e-05, "loss": -1990.978, "head_losses": [242.497, -726.34, 8.62, 224.479, -997.567, -754.47, 5.907, 5.896]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:55:42,371", "type": "train", "epoch": 138, "batch": 154, "n_batches": 1080, "time": 0.711, "data_time": 0.0, "lr": 1e-05, "loss": -440.91, "head_losses": [261.086, -191.182, 9.195, 252.347, -476.207, -309.306, 6.581, 6.576]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:55:49,642", "type": "train", "epoch": 138, "batch": 165, "n_batches": 1080, "time": 0.688, "data_time": 0.0, "lr": 1e-05, "loss": -776.19, "head_losses": [227.267, -340.846, 8.224, 209.617, -552.084, -341.391, 6.515, 6.508]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:55:56,984", "type": "train", "epoch": 138, "batch": 176, "n_batches": 1080, "time": 0.718, "data_time": 0.0, "lr": 1e-05, "loss": -1843.319, "head_losses": [303.442, -827.1, 5.143, 250.792, -874.616, -708.506, 3.763, 3.763]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:56:04,811", "type": "train", "epoch": 138, "batch": 187, "n_batches": 1080, "time": 0.716, "data_time": 0.0, "lr": 1e-05, "loss": -526.815, "head_losses": [250.192, -353.142, 7.226, 213.732, -407.395, -249.127, 5.846, 5.853]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:56:12,532", "type": "train", "epoch": 138, "batch": 198, "n_batches": 1080, "time": 0.726, "data_time": 0.0, "lr": 1e-05, "loss": -2074.082, "head_losses": [192.724, -829.948, 6.376, 169.985, -920.939, -702.556, 5.14, 5.137]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:56:19,816", "type": "train", "epoch": 138, "batch": 209, "n_batches": 1080, "time": 0.724, "data_time": 0.0, "lr": 1e-05, "loss": -1061.801, "head_losses": [321.626, -516.477, 8.733, 295.665, -691.82, -492.492, 6.486, 6.478]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:56:26,930", "type": "train", "epoch": 138, "batch": 220, "n_batches": 1080, "time": 0.687, "data_time": 0.0, "lr": 1e-05, "loss": -1430.217, "head_losses": [236.696, -572.359, 8.369, 207.166, -756.222, -565.628, 5.884, 5.878]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:56:34,728", "type": "train", "epoch": 138, "batch": 231, "n_batches": 1080, "time": 0.689, "data_time": 0.0, "lr": 1e-05, "loss": -98.27, "head_losses": [446.626, -189.861, 10.162, 359.669, -462.71, -277.133, 7.492, 7.485]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:56:42,235", "type": "train", "epoch": 138, "batch": 242, "n_batches": 1080, "time": 0.407, "data_time": 0.0, "lr": 1e-05, "loss": 1680.333, "head_losses": [462.434, 341.882, 15.756, 427.863, 50.138, 357.962, 12.148, 12.15]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:56:49,524", "type": "train", "epoch": 138, "batch": 253, "n_batches": 1080, "time": 0.388, "data_time": 0.0, "lr": 1e-05, "loss": -1169.699, "head_losses": [183.5, -478.075, 6.325, 171.865, -589.427, -471.877, 3.995, 3.995]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:56:57,261", "type": "train", "epoch": 138, "batch": 264, "n_batches": 1080, "time": 0.657, "data_time": 0.0, "lr": 1e-05, "loss": -708.93, "head_losses": [419.319, -507.892, 6.167, 334.909, -549.143, -422.282, 4.999, 4.993]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:57:05,071", "type": "train", "epoch": 138, "batch": 275, "n_batches": 1080, "time": 0.661, "data_time": 0.0, "lr": 1e-05, "loss": -787.568, "head_losses": [351.617, -399.401, 12.749, 338.605, -679.579, -429.108, 8.778, 8.77]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:57:12,480", "type": "train", "epoch": 138, "batch": 286, "n_batches": 1080, "time": 0.725, "data_time": 0.0, "lr": 1e-05, "loss": -1001.895, "head_losses": [264.949, -487.542, 6.298, 247.594, -611.526, -430.479, 4.408, 4.402]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:57:19,678", "type": "train", "epoch": 138, "batch": 297, "n_batches": 1080, "time": 0.648, "data_time": 0.0, "lr": 1e-05, "loss": -799.337, "head_losses": [169.984, -357.481, 3.536, 147.476, -455.641, -313.148, 2.968, 2.969]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:57:27,569", "type": "train", "epoch": 138, "batch": 308, "n_batches": 1080, "time": 0.717, "data_time": 0.0, "lr": 1e-05, "loss": -2266.953, "head_losses": [211.284, -905.543, 8.209, 192.214, -967.535, -817.477, 5.95, 5.944]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:57:35,486", "type": "train", "epoch": 138, "batch": 319, "n_batches": 1080, "time": 0.724, "data_time": 0.0, "lr": 1e-05, "loss": -227.813, "head_losses": [297.949, -375.425, 8.893, 294.855, -303.38, -165.513, 7.41, 7.398]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:57:42,171", "type": "train", "epoch": 138, "batch": 330, "n_batches": 1080, "time": 0.746, "data_time": 0.0, "lr": 1e-05, "loss": 530.408, "head_losses": [492.089, -121.032, 5.431, 336.663, -158.379, -32.933, 4.282, 4.287]}
{"message": null, "levelname": "INFO", "name": "openpifpaf.network.trainer", "asctime": "2021-09-15 14:57:49,341", "type": "train", "epoch": 138, "batch": 341, "n_batches": 1080, "time": 0.387, "data_time": 0.0, "lr": 1e-05, "loss": -2070.688, "head_losses": [444.346, -953.638, 10.076, 349.02, -1081.036, -853.928, 7.237, 7.233]}
```

I visiualized the logs file,
and the loss-epoch image looks like this:

![0005](https://user-images.githubusercontent.com/25741435/133397075-fcccefb6-691e-4e87-9e82-2b26f63b5aba.jpeg)

and:

![0004](https://user-images.githubusercontent.com/25741435/133397486-a6373c3c-dce2-4ac5-ab91-44189ce5821b.jpeg)

![0007](https://user-images.githubusercontent.com/25741435/133397496-6b98c6af-5646-4f8b-b242-d407bb08a4c7.jpeg)

What i am trying now are:

1. change the square_size, my images in dataset are 1600 * 900, so i'm trying to set this bigger. Due to my GPU limit, i can now set square_size to 721(crowdpose default is 385), but by the time i'm writing this issue, the new training also get negative losses. (change square_size will work ? i'm not sure

2. do training without data argument. (seems it doesn't work

3. do some test whatever. ( I am not working on this yet.

I am not sure whatever this situation is acceptable or there are some bad properties in my dataset. My dataset is generated by computer games so it is a synthetic dataset, but i just cannot show you right now due to some reasons.

I am also reading your paper now and try to understand the structure, it's a little bit hard for me now for i'm new to this field, so i am looking for some help QwQ, thanks.

**Comment by svenkreiss** [comment]
Hi @yyccli. Negative losses are expected. All your loss graphs look good to me. Feel free to re-open if there is another issue.
